### PR TITLE
Fixed IntelliSense suggestion on typing dot after method group

### LINKED ISSUE #8321: Weird completion list upon dotting into the name of a static method
1. Ctrl-J at the caret. Why are x, args, z, and instance methods in the list?
   
   ```
   class Program
   {
   int x;
   
   static void Main(string[] args, int[] z)
   {
       Main.$$
   }
   }
   ```

### LINKED ISSUE #16069: Typing dot after method group incorrectly shows IntelliSense
**Version Used**: 2017 RC Refresh

```C# 
class C {
	void M<T>() {M<C>. }
	void M() {M.}
}
```

Typing either `.` in this code will open IntelliSense, even though no identifier can go there.

_pedantic_

### PR DESCRIPTION
**Pull request by zaytsev-victor**
**Customer scenario**
Typing dot after method group incorrectly shows IntelliSense

**Bugs this fixes:**
Fixes #8321
Fixes #16069

**Risk**
Low

**Performance impact**
Low

**Is this a regression from a previous update?**
No

**How was the bug found?**
customer reported

### CODE REVIEW

**Comment by sharwell** [comment]
:bulb: As a GitHub "pro-tip", you can use the following text in the description when you send a PR to indicate that the PR should actually close the linked issue when it gets merged:

```
Fixes #8321
```

If you just put the issue reference by itself (without the word `Fixes` in front), it still links the pull request to the issue but doesn't close the issue.

(Seems relevant since you appear to want to keep fixing all the things 👍 )

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs [hunk 1]
```diff
@@ -458,12 +458,19 @@ private static ImmutableArray<ISymbol> GetSymbolsForNamespaceDeclarationNameCont
...
                 // If the thing on the left is a lambda expression, we shouldn't show anything.
+                var originalExpressionKind = originalExpression.Kind();
                 if (symbol.Kind == SymbolKind.Method &&
                     ((IMethodSymbol)symbol).MethodKind == MethodKind.AnonymousFunction)
                 {
                     return ImmutableArray<ISymbol>.Empty;
                 }
 
+                if (symbol.Kind == SymbolKind.Method &&
+                    (originalExpressionKind == SyntaxKind.IdentifierName || originalExpressionKind == SyntaxKind.GenericName))
```
Comment (sharwell) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs]: :bulb: Assuming the two-`SyntaxKind` overload is available at this location:

```
originalExpression.IsKind(SyntaxKind.IdentifierName, SyntaxKind.GenericName)
```

Also, can you include an example of why the previous code wouldn't work? I think seeing an example may help me avoid making a similar mistake in the future and having to go track it down. :smile:

Comment (zaytsev-victor) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs]: @sharwell, it seems that I forgot to build project before running tests when I tried to simplify the fix :)

example of why the previous code wouldn't work:
```csharp
class C
{
    string M()
    {
        M().$$
        return string.Empty;
    }
}
```
no suggestion with the previous fix here.

Comment (zaytsev-victor) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs]: + ```IsKind``` method isn't available at this location

**Comment by Pilchie** [comment]
Merging - thanks for the contribution!

**Pull request merged**